pkg/testhelper/accessory.go

```diff
@@ -268,14 +261,37 @@ func GetFreePort(t TestingTInterface) string {
 		if err != nil {
 			t.Fatalf("could not listen on free port: %v", err)
 		}
-		defer func() {
-			if err := l.Close(); err != nil {
+		defer func(c io.Closer) {
+			if err := c.Close(); err != nil {
 				t.Errorf("could not close listener: %v", err)
 			}
-		}()
+		}(l)
```

Comment:
This is technically unrelated but it could leak unclosed connections as written today.